### PR TITLE
DeleteSnapshot implementation in SnapshotManager

### PR DESCRIPTION
**Pull request by deepakkinni**
1. Implementation in SnapshotManager.
2. Add deletesnapshots into deployment yamls.
3. Fix type backpRepository -> backupRepository
4. Add snapshotReference into the upload CRD (ran make update to generate files)
5. Fixed Snapshot api in paravirt pe to return the snap-id
6. Additional logging.
7. End-to-end test to verify DeleteSnapshot.

Pending:
1. Update go.mod with correct version of astrolabe after PR 158 is merged.
2. Re-run 'make update' if PR 170 is submitted earlier.

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/170/

File: pkg/snapshotmgr/snapshot_manager.go [hunk 1]
```diff
@@ -233,25 +235,24 @@ func (this *SnapshotManager) createSnapshot(peID astrolabe.ProtectedEntityID, ta
 		}
 		return true, nil
 	})
-	this.Debugf("Return from the call of astrolabe Snapshot API for PE %s", peID.String())
+	this.Infof("Return from the call of astrolabe Snapshot API for PE %s", peID.String())
```
[the remark this answers is not in the document]
Comment (deepakkinni) [reply pkg/snapshotmgr/snapshot_manager.go:240]: I am just changing it temporarily until we have a go from QE, will downgrade this and several other log statements afterwards (https://jira.eng.vmware.com/browse/DPCP-286) . I've found additional logging really helpful.

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/177/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/178/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/179/

**Pull request merged**